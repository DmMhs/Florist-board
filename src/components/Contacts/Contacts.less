.Contacts {
  display: flex;
  flex-direction: column;
  align-items: center;
  .contacts-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    h2 {
      font-size: 2rem;
      font-weight: normal;
      color: #333333;
    }
    hr {
      width: 75%;
      border: 0.5px solid #eee;
      margin: 1.5rem 0;
    }
  }
}
@media screen and (max-width: 768px) {
  .Contacts {
    margin-top: 4rem;
  }
}
