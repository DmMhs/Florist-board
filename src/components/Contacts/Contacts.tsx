import React from 'react';

const contacts = () => {
    return (
        <div className="Contacts">
        Contacts Component         
        </div>
    );
};

export default contacts;