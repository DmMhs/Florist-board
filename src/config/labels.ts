const labels = {
  brand: 'FLORIST.UA',
  adress: 'MARKET SQUARE, LVIV, UA',
  postCode: '346346',
  phone: '+123-45-678',
  navigation: {
    home: 'HOME',
    shop: 'SHOP',
    gallery: 'GALLERY',
    contacts: 'CONTACTS',
    cart: 'CART',
    account: {
      main: 'MY ACCOUNT',
      menu: {
        signIn: 'SIGN IN',
        signUp: 'SIGN UP'
      }
    }
  }
};

export default labels;
