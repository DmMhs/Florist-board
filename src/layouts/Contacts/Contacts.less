.Contacts {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: calc(100vh - 7rem);
  .contacts-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    h2 {
      font-size: 2rem;
      font-weight: normal;
      color: #333333;
    }
    hr {
      width: 75%;
      border: 0.5px solid #eee;
      margin: 1.5rem 0;
    }
  }
}
@media screen and (max-width: 768px) {
  .Contacts {
    margin-top: 2.5rem;
  }
}
