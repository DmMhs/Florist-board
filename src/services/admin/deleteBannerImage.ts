import { homeImagesRef, storageRef } from '../../firebase';

const getKeyByValue = (object: { [key: string]: string }, value: string) => {
  return Object.keys(object).find(key => object[key] === value);
};

const map = { first: '1', second: '2' };

export const deleteBannerImage = (imageName: string, imageUrl: string) => {
  const confirm = window.confirm('Are you sure?');
  if (confirm) {
    return storageRef
      .child('banner')
      .child(imageName)
      .delete()
      .then(response => {
        return homeImagesRef
          .once('value')
          .then(snapshot => {
            const imageKey = getKeyByValue(snapshot!.val(), imageUrl);
            return homeImagesRef.child(imageKey as string).remove();
          })
          .catch(err => console.log(err));
      });
  }
};
