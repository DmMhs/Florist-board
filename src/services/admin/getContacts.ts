import { database } from '../../firebase';

export const getContacts = () => {
  return database
    .ref()
    .child('contacts')
    .once('value')
    .then(snapshot => {
      return snapshot.val();
    });
};
