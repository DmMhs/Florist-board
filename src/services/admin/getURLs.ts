import { database } from '../../firebase';

export const getURLs = async () => {
  return database
    .ref()
    .child('urls')
    .once('value')
    .then(snapshot => {
      return snapshot!.val();
    });
};
