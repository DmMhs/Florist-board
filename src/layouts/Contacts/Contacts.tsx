import React, { Component } from 'react';

import GoogleMap from './Map/GoogleMap';
import ContactInfo from './ContactInfo/ContactInfo';
import { AppContext } from '../../AppContext';
import { urls } from '../../config/urls';
import './Contacts.less';
import { getURLs } from '../../services/admin/getURLs';
import { getContacts } from '../../services/admin/getContacts';

interface ContactsProps {}
interface ContactsState {
  contacts: {
    en: {
      address: string;
    };
    ua: {
      address: string;
    };
    postCode: string;
    phone: string;
    email: string;
  };
  social_urls: {
    facebook: string;
    instagram: string;
    telegram: string;
  };
  google_map_url: string;
  fetchInProgress: boolean;
}

class Contacts extends Component<ContactsProps, ContactsState> {
  constructor(props: ContactsProps) {
    super(props);
    this.state = {
      contacts: {
        en: {
          address: ''
        },
        ua: {
          address: ''
        },
        postCode: '',
        phone: '',
        email: ''
      },
      social_urls: {
        facebook: '',
        instagram: '',
        telegram: ''
      },
      google_map_url: '',
      fetchInProgress: true
    };
  }

  public componentDidMount = async () => {
    const contacts = await getContacts();

    this.setState({
      contacts: contacts
    });
    const data = await getURLs();
    this.setState({
      google_map_url: data.google_map_address,
      social_urls: {
        facebook: data.socials.facebook,
        instagram: data.socials.instagram,
        telegram: data.socials.telegram
      },
      fetchInProgress: false
    });
  };

  public render() {
    const context = this.context;
    const labels = context.state.labels;
    const lang = context.state.lang;
    const contacts = this.state.contacts;
    const social = this.state.social_urls;
    return (
      <AppContext.Consumer>
        {value => (
          <div className="Contacts">
            <div className="contacts-wrapper">
              <h2>{labels[lang].pages.contacts.map}</h2>
              <GoogleMap url={urls.google_map_address} />
              <hr />
              <ContactInfo
                address={(contacts as { [key: string]: any })[lang].address}
                phone={contacts.phone}
                email={contacts.email}
                instagram={social.instagram}
                facebook={social.facebook}
                telegram={social.telegram}
              />
            </div>
          </div>
        )}
      </AppContext.Consumer>
    );
  }
}

Contacts.contextType = AppContext;

export default Contacts;
