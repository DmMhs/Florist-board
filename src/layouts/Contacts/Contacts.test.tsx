import React from 'react';
import { shallow, mount } from 'enzyme';

import Contacts from './Contacts';
import { BrowserRouter } from 'react-router-dom';
import AppContextProvider from '../../AppContext';
import labels from '../../config/labels';
import * as getContactsFunction from '../../services/admin/getContacts';
import * as getURLsFunction from '../../services/admin/getURLs';

describe('Contacts component works as expected', () => {
  it('matches a snapshot', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <Contacts />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('contacts are displayed', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <Contacts />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper.find('.Contacts').exists()).toBeTruthy();
  });
  // it('function was called', async () => {
  //   const p = Promise.resolve();
  //   const spyGetContacts = jest
  //     .spyOn(getContactsFunction, 'getContacts')
  //     .mockImplementation(() => p);

  //   const wrapper = mount(
  //     <BrowserRouter>
  //       <AppContextProvider>
  //         <Contacts />
  //       </AppContextProvider>
  //     </BrowserRouter>
  //   );
  //   const context = wrapper.find('AppContextProvider').instance();
  //   context.setState({
  //     lang: 'en',
  //     labels: labels,
  //     fetchInProgress: false,
  //     mobileMode: true,
  //     showNavigation: false,
  //     togglePosition: 'absolute'
  //   });
  //   wrapper.update();
  //   await p;

  //   expect(spyGetContacts).toHaveBeenCalled();
  // });
});
// 89.57 |    83.44 |    84.39 |    89.56
// 89.29 |    83.44 |    83.97 |    89.27
