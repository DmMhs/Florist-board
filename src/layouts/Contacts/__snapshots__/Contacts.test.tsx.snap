// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts component works as expected matches a snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <Contacts>
            <div
              className="Contacts"
            >
              <div
                className="contacts-wrapper"
              >
                <h2>
                  Where to find us?
                </h2>
                <GoogleMap
                  url="https://maps.google.com/maps?q=Lviv%2C%20Rynok%20Square&t=&z=13&ie=UTF8&iwloc=&output=embed"
                >
                  <div>
                    <WithStyles(CircularIndeterminate)>
                      <CircularIndeterminate
                        classes={
                          Object {
                            "progress": "CircularIndeterminate-progress-1",
                          }
                        }
                      >
                        <div
                          className="Spinner"
                        >
                          <WithStyles(CircularProgress)
                            className="CircularIndeterminate-progress-1"
                          >
                            <CircularProgress
                              className="CircularIndeterminate-progress-1"
                              classes={
                                Object {
                                  "circle": "MuiCircularProgress-circle-8",
                                  "circleDisableShrink": "MuiCircularProgress-circleDisableShrink-11",
                                  "circleIndeterminate": "MuiCircularProgress-circleIndeterminate-10",
                                  "circleStatic": "MuiCircularProgress-circleStatic-9",
                                  "colorPrimary": "MuiCircularProgress-colorPrimary-5",
                                  "colorSecondary": "MuiCircularProgress-colorSecondary-6",
                                  "indeterminate": "MuiCircularProgress-indeterminate-4",
                                  "root": "MuiCircularProgress-root-2",
                                  "static": "MuiCircularProgress-static-3",
                                  "svg": "MuiCircularProgress-svg-7",
                                }
                              }
                              color="primary"
                              disableShrink={false}
                              size={40}
                              thickness={3.6}
                              value={0}
                              variant="indeterminate"
                            >
                              <div
                                className="MuiCircularProgress-root-2 MuiCircularProgress-colorPrimary-5 MuiCircularProgress-indeterminate-4 CircularIndeterminate-progress-1"
                                role="progressbar"
                                style={
                                  Object {
                                    "height": 40,
                                    "width": 40,
                                  }
                                }
                              >
                                <svg
                                  className="MuiCircularProgress-svg-7"
                                  viewBox="22 22 44 44"
                                >
                                  <circle
                                    className="MuiCircularProgress-circle-8 MuiCircularProgress-circleIndeterminate-10"
                                    cx={44}
                                    cy={44}
                                    fill="none"
                                    r={20.2}
                                    strokeWidth={3.6}
                                    style={Object {}}
                                  />
                                </svg>
                              </div>
                            </CircularProgress>
                          </WithStyles(CircularProgress)>
                        </div>
                      </CircularIndeterminate>
                    </WithStyles(CircularIndeterminate)>
                    <div
                      className="mapouter Map"
                      style={
                        Object {
                          "height": "450px",
                          "position": "relative",
                          "textAlign": "right",
                          "width": "724px",
                        }
                      }
                    >
                      <div
                        className="gmap_canvas"
                        style={
                          Object {
                            "background": "none!important",
                            "height": "450px",
                            "overflow": "hidden",
                            "width": "724px",
                          }
                        }
                      >
                        <iframe
                          frameBorder="0"
                          height="450"
                          id="gmap_canvas"
                          marginHeight={0}
                          marginWidth={0}
                          onLoad={[Function]}
                          scrolling="no"
                          src="https://maps.google.com/maps?q=Lviv%2C%20Rynok%20Square&t=&z=13&ie=UTF8&iwloc=&output=embed"
                          width="724px"
                        />
                      </div>
                    </div>
                  </div>
                </GoogleMap>
                <hr />
                <contactInfo
                  address=""
                  email=""
                  facebook=""
                  instagram=""
                  phone=""
                  telegram=""
                >
                  <div
                    className="ContactsInfo"
                  >
                    <h3>
                      Feel free to get in touch with us!
                    </h3>
                    <hr />
                    <div
                      className="contact-option"
                    >
                      <div
                        className="icon-wrapper"
                      >
                        <img
                          alt="address"
                          className="address-icon"
                          src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Faddress.png?alt=media&token=2fb28441-7270-4ebe-93eb-54bf0c7f1dfa"
                        />
                      </div>
                       
                    </div>
                    <div
                      className="contact-option"
                    >
                      <div
                        className="icon-wrapper"
                      >
                        <img
                          alt="phone"
                          className="phone-icon"
                          src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Fmobile-phone.png?alt=media&token=a184bbb9-5b5b-47a9-bb33-d61740e95b9c"
                        />
                      </div>
                       
                    </div>
                    <div
                      className="contact-option"
                    >
                      <div
                        className="icon-wrapper"
                      >
                        <img
                          alt="email"
                          className="email-icon"
                          src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Femail.png?alt=media&token=0e246b98-8137-4068-9594-d9441b571a06"
                        />
                      </div>
                       
                    </div>
                    <div
                      className="socials"
                    >
                       
                      <hr />
                      <div
                        className="contact-option"
                      >
                        <div
                          className="icon-wrapper"
                        >
                          <a
                            href=""
                          >
                            <img
                              alt="facebook"
                              className="facebook-icon"
                              src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Ffacebook.png?alt=media&token=5fdb89e0-3382-43af-af56-379ed4a5db2f"
                            />
                          </a>
                        </div>
                      </div>
                      <div
                        className="contact-option"
                      >
                        <div
                          className="icon-wrapper"
                        >
                          <a
                            href=""
                          >
                            <img
                              alt="instagram"
                              className="instagram-icon"
                              src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Finstagram.png?alt=media&token=58c1731f-94f2-4d23-82e8-1e2d753413b7"
                            />
                          </a>
                        </div>
                      </div>
                      <div
                        className="contact-option"
                      >
                        <div
                          className="icon-wrapper"
                        >
                          <a
                            href=""
                          >
                            <img
                              alt="telegram"
                              className="telegram-icon"
                              src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Ftelegram.png?alt=media&token=95badfef-90f7-49b8-858e-a92a03657f4e"
                            />
                          </a>
                        </div>
                         
                      </div>
                    </div>
                  </div>
                </contactInfo>
              </div>
            </div>
          </Contacts>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
